### PR TITLE
Library Folder Pagination

### PR DESCRIPTION
**Pull request by OlegZharkov**
current status:
everything should work

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/buildapp.py [hunk 1]
```diff
@@ -985,6 +985,12 @@ def connect_invocation_endpoint(endpoint_name, endpoint_suffix, action, conditio
                           action='get_permissions',
                           conditions=dict(method=["GET"]))
 
+    webapp.mapper.connect('search_folder_items',
+                          '/api/folders/{encoded_folder_id}/search',
+                          controller='folders',
```
Comment (dannon) [inline lib/galaxy/webapps/galaxy/buildapp.py]: folder_contents

**Comment by OlegZharkov** [comment]
I would appreciate, if someone can try this out. (especially, python experts)

**Comment by jmchilton** [comment]
If I hit the library list from the main GUI, I land up at:

http://127.0.0.1:8080/library/list

and click a folder, I get here:

http://127.0.0.1:8080/library/folders/Ffc49e35ec7bb510b

then if I hit ".."

I get to 

http://127.0.0.1:8080/library/list

and then clicking a folder is broken because that slash is being copied into the folders URL:

http://127.0.0.1:8080/library/list/folders/Ffc49e35ec7bb510b

**Comment by OlegZharkov** [comment]
> If I hit the library list from the main GUI, I land up at:
> 
> http://127.0.0.1:8080/library/list
> 
> and click a folder, I get here:
> 
> http://127.0.0.1:8080/library/folders/Ffc49e35ec7bb510b
> 
> then if I hit ".."
> 
> I get to
> 
> http://127.0.0.1:8080/library/list
> 
> and then clicking a folder is broken because that slash is being copied into the folders URL:
> 
> http://127.0.0.1:8080/library/list/folders/Ffc49e35ec7bb510b

Hi John, 
It's fixed in https://github.com/galaxyproject/galaxy/pull/10284

**Comment by jmchilton** [comment]
Ahh that is why it wasn't a problem in 20.09, alright thanks!

**Comment by jmchilton** [comment]
Awesome stuff by the way, thanks so much for solving this big problem! Very nice.

**Comment by OlegZharkov** [comment]
Thanks a lot! @jmchilton @dannon @mvdbeek 
I cannot believe, that this project is finally finished :smile:

**Pull request merged**